Java/core/src/main/java/com/amazon/randomcutforest/store/PointStore.java

```diff
@@ -190,61 +208,55 @@ public int decrementRefCount(int index) {
      * @throws IllegalStateException    if the point store is full.
      */
     public int add(double[] point, long sequenceNum) {
-        checkArgument(point.length == dimensions, "point.length must be equal to dimensions");
-
-        if (shingleSize > 1 && shingleAwareOverlapping) {
-            /**
-             * corresponds to the shingled/overlapped case. Currently we are considering a
-             * perfect sequence where the shift corresponds to one time unit and that
-             * represents baseDimension new values. Note that if the alignment test fails,
-             * then the points are still copied -- in future it may be reasonable to check
-             * for different alignments -- which would be beneficial for larger shingles. In
-             * extreme a segment tree type data structure could also be used.
-             */
-            if (startOfFreeSegment > capacity * dimensions - baseDimension) {
-                // the above ensures that the most recent values in a shingle can be writtem
-                compact();
-            }
-
-            boolean test = (startOfFreeSegment - dimensions + baseDimension >= 0);
-            if (test) {
-                test = checkShingleAlignment(startOfFreeSegment, point);
-            }
-            if (test && startOfFreeSegment + baseDimension <= capacity * dimensions) {
-                int nextIndex = takeIndex(); // no more compactions
-                refCount[nextIndex] = 1; // has to be after compactions
-                locationList[nextIndex] = startOfFreeSegment - dimensions + baseDimension;
-                copyPoint(point, dimensions - baseDimension, startOfFreeSegment, baseDimension);
-                startOfFreeSegment += baseDimension;
-                return nextIndex;
+        checkArgument(internalShinglingEnabled || point.length == dimensions,
+                "point.length must be equal to dimensions");
+        checkArgument(!internalShinglingEnabled || point.length == baseDimension,
+                "point.length must be equal to dimensions");
+
+        double[] tempPoint = point;
+        if (internalShinglingEnabled) {
+            tempPoint = changeShingleInPlace(internalShingle, point);
+            nextTimeStamp++;
```

Comment:
What is a timestamp and how does it relate to sequence number?

Reply:
It is the sequence number - for internal sequence number we keep track of the value seen and check it the next add() aligns.